### PR TITLE
XWIKI-21492: Underline inline links

### PR DESCRIPTION
**Pull request by Sereza7**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-21492
# Changes

## Description

* Fixed the xpaths to account for the new preference category

## Clarifications

* Adding the `accessibility settings` section changed some of the xpaths (that are not very robust, they rely on order only, but unfortunately there's no other easy way)
* The selectors on the `PreferencesEditPage` page are not broken because their definition is more sturdy :)

# Screenshots & Video

None, development changes only.

# Executed Tests

None.
I don't have a build with the changes from XWIKI-21492 yet. However the issue here is quite easy to understand and fix.

Fixes https://ci.xwiki.org/job/XWiki%20Environment%20Tests/job/xwiki-platform/job/master/527/testReport/org.xwiki.user.test.ui/AllIT$NestedUserProfileIT/Oracle_19_3_0_se2__Jetty_10_jdk21__Firefox___Docker_tests_for_xwiki_platform_user_profile___Build_for_Oracle_19_3_0_se2__Jetty_10_jdk21__Firefox___Docker_tests_for_xwiki_platform_user_profile___changeUserProfile/ and https://ci.xwiki.org/job/XWiki%20Environment%20Tests/job/xwiki-platform/job/master/527/testReport/org.xwiki.user.test.ui/AllIT$NestedUserProfileIT/Oracle_19_3_0_se2__Jetty_10_jdk21__Firefox___Docker_tests_for_xwiki_platform_user_profile___Build_for_Oracle_19_3_0_se2__Jetty_10_jdk21__Firefox___Docker_tests_for_xwiki_platform_user_profile___changeDefaultEditor/

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * None

### CODE REVIEW

**Comment by Sereza7** [comment]
For reference, here is a screenshot of the structure the xpath are navigating into:
image

File: xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java [hunk 1]
```diff
@@ -33,13 +33,13 @@ public class PreferencesUserProfilePage extends AbstractUserProfilePage
     @FindBy(xpath = "//div[@id='preferencesPane']//div[@class='editProfileCategory']/a")
     private WebElement editPreferences;
 
-    @FindBy(xpath = "//div[@id='preferencesPane']/div[1]/div/dl[2]/dd[1]")
+    @FindBy(xpath = "//div[@id='preferencesPane']/div[1]/div/dl[3]/dd[1]")
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java:36]: Why keeping fragile and hard to read selectors like this? 
Afaics all the `dd` elements have a nice `data-user-property="TYPE"` that we can use to define much better selectors.

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java:36]: I'm not sure how to test this exactly yet (without spending half an hour applying the changes from XWIKI-21492 manually). So I stuck with the simplest solution that I knew would not create a regression. I'm not super comfortable with xpaths yet so I'd want to test if I made a larger change.
If we don't merge this right now, I'll wait for the updated build tomorrow, test with it and update the PR then.

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java:36]: You can quite easily build the changes locally and run the tests again.
Something like `mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker` should allow you to test you change (assuming your branch is up-to-date with master).

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java:36]: You can even add the `-U` option to also update unrelated dependencies, but with the set of modules listed in `-pl`, all the relevant bits should be ok `mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker`
Also, if you want to speed up the docker validation part, you can filter on a given test:
`mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker -Dit.test=MyClass#mymethod`.
Finally, there is also an option to run the test against a server running on localhost:8080 (see https://dev.xwiki.org/xwiki/bin/view/Community/Testing/DockerTesting/#HExistingXWikiinstance)

**Review by manuelleduc — APPROVED** [review]

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less [hunk 2]
```diff
@@ -162,10 +162,11 @@ body.content {
...
     #xwikicontent {
-      &.xwiki-livetable-container,
-      &.xtree,
-      &.xwikitabbar,
-      &.buttonwrapper {
+      & .xwiki-livetable-container,
+      & .xtree,
+      & .xwikitabbar,
+      & .buttonwrapper,
+      & #user-menu-col {
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less:169]: Disable account is still underlined in the user profile

image

cc @michitux for that part as you followed the topic more than I did.

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/general.less:169]: image

Okay, as of now the action links (that often look like a button) are only addressed by `& .buttonwrapper` which takes care of all the fake buttons in [inline forms](https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/FrontendResources/InlineForms/). 
As seen in the screenshot, this link is not contained in a `.buttonwrapper` so it's not matched with our selector.
An improvement would be to match on `.btn` itself. I'll keep track of this comment and start a fix PR with this fix along some other we might find along the way next week (to hopefully get it in the full release, should be very safe to merge :) ).

File: xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm [hunk 3]
```diff
@@ -148,10 +148,14 @@
...
 ## We put a space before each class added in #if, to avoid losing them because of velocity line stripping
+## TODO: the classes should be refactored to use a array of classes where individal classes are happened on by one and
+## joined with space at the end
+## Note: when showLeftPanels and showRightPanels are both equals to 0, it is intentional of hideleft and hideright to
+## be concatenated without space.
 <$bodyTagName id="body" class="skin-flamingo wiki-${xcontext.database} ${bodyAction}body panel-left-width-${leftPanelsWidth} panel-right-width-${rightPanelsWidth} drawer drawer--right ${preferenceUnderlining}
   #if("$!doc.space" != "") space-${escapetool.xml($doc.space.replaceAll(' ', '_'))}#end
   #if($showLeftPanels == "0") hideleft#end
-  #if($showRightPanels == "0") hideright#end
+  #if($showRightPanels == "0")hideright#end
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm:158]: I don't have the impression that this fixed anything. The left column is still shown on the login form.

Comment (manuelleduc) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm:158]: Or, maybe even clearer, `#if($showLeftPanels == "0") hideleft#end#if($showRightPanels == "0")hideright#end` both in the same line

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/htmlheader.vm:158]: I did remove it in https://github.com/xwiki/xwiki-platform/pull/3074/commits/3c1fcc7505bd411258bc57c135a6f4464c930f43 but somehow it got added back before the merge :/

**Pull request merged**